app/controllers/api/v1/users_controller.rb

```diff
@@ -29,6 +29,57 @@ def update
         end
       end
 
+      # exist
+      #
+      # When given a user's email, checks if that user is already registered in the database
+      # and sends the proper redirect url to the frontend accordingly. Called when logging in
+      # via social media.
+      # @param params The data passed in from the frontend, from which [:user][:email] (the user's email) is extracted. Expecting :user in params with parameter :email inside.
+
+      def exist
+        user = User.find_by(email: params[:user][:email])
+        Rails.logger.info "************ user is = #{user}"
+        @redirect_path = '/profile' #change this to actually redirect to the social login function
+        unless user
+          @redirect_path = '/additional_info'
+        end
+        Rails.logger.info "!!!!!!!!!!!! path is #{@redirect_path}"
+        render json: { redirect_to: @redirect_path }
+      end
+
+      # social
+      #
+      # When using social logins, creates the user in the database if necessary, then logs them in
+      # and sends the proper redirect url to the frontend accordingly.
+
+      def social
```

Comment:
You wanna give another stab at this Yard doc, based on the above?

Reply:
Yep, I've revised the docs!